dotnet/Simple.Sauce/SauceOptions.cs

```diff
@@ -55,83 +49,125 @@ public SauceOptions()
-                                                edgeVersion);
-            ConfiguredEdgeOptions.BrowserVersion = edgeVersion.Value;
-            ConfiguredEdgeOptions.PlatformName = DEFAULT_PLATFORM;
+            var options = JsonConvert.SerializeObject(this, JsonUtils.SerializerSettings());
+            var sauceOptions = JsonConvert.DeserializeObject<Dictionary<string, object>>(options, JsonUtils.SerializerSettings());
+
+            return sauceOptions;
         }
 
-        public void WithChrome()
+        public SauceOptions WithChrome() => WithChrome(DEFAULT_BROWSER_VERSION, DEFAULT_PLATFORM);
+        public SauceOptions WithChrome(string version) => WithChrome(version, DEFAULT_PLATFORM);
+        public SauceOptions WithChrome(string version, string platform)
         {
-            ConfiguredChromeOptions.BrowserVersion = DEFAULT_BROWSER_VERSION;
-            ConfiguredChromeOptions.PlatformName = DEFAULT_PLATFORM;
+            ConfiguredOptions = new ChromeOptions
+            {
+                BrowserVersion = version,
+                PlatformName = platform
+            };
+
+            return this;
         }
 
-        public void WithChrome(string chromeVersion)
+        public SauceOptions WithEdge() => WithEdge(EdgeVersion.Latest, DEFAULT_PLATFORM);
+        public SauceOptions WithEdge(EdgeVersion version) => WithEdge(version, DEFAULT_PLATFORM);
+        public SauceOptions WithEdge(EdgeVersion edgeVersion, string platform)
         {
-            ConfiguredChromeOptions.BrowserVersion = chromeVersion;
+            if (edgeVersion is null)
+                throw new ArgumentNullException("Please supply a valid Edge Version. You supplied an invalid value => " +
+                                                edgeVersion);
+
+            ConfiguredOptions = new EdgeOptions
+            {
+                BrowserVersion = edgeVersion.Value,
+                PlatformName = platform
+            };
+
+            return this;
         }
 
-        public void WithSafari()
+        public SauceOptions WithFirefox() => WithFirefox(DEFAULT_BROWSER_VERSION, DEFAULT_PLATFORM);
+        public SauceOptions WithFirefox(string version) => WithFirefox(version, DEFAULT_PLATFORM);
+        public SauceOptions WithFirefox(string version, string platform)
         {
-            WithSafari(DEFAULT_BROWSER_VERSION);
+            ConfiguredOptions = new FirefoxOptions
+            {
+                BrowserVersion = version,
+                PlatformName = platform
+            };
+
+            return this;
         }
 
-        public void WithSafari(string safariVersion)
+        public SauceOptions WithSafari() => WithSafari(DEFAULT_BROWSER_VERSION);
+
+        public SauceOptions WithSafari(string version)
         {
-            ConfiguredSafariOptions.BrowserVersion = safariVersion;
-            ConfiguredSafariOptions.PlatformName = MatchCorrectPlatformToBrowserVersion(safariVersion);
+            var platform =
```

Comment:
Codacy found an issue: [Remove the unused local variable 'platform'.](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6342070)